### ISSUE TITLE
CS: GO static props scaled visually but not physically

### ISSUE THREAD

**Issue by Fennecai** [Facepunch/garrysmod-issues#4500]
### Details
I am running version 2020.5.11.

When I tried to play the CS: GO map "dz_sirocco" in single-player, I ran into some issues with strange invisible walls. specifically, there is a well with stairs spiraling down into it on the map; and it is in this well and it's surrounding underground tunnels which i encounter the issues the most. when i first found the invisible walls, i assumed some weird issue with brushes in the map, and i decompiled the map to attempt a fix, but that's when i noticed in hammer that all these walls were actually the collision meshes of nearby metal pipe models which had been scaled down in CS:GO's hammer. in game, the pipes appear correctly- but the phycisal collisions on them are at the normal, huge size (and i suspect are also rotated incorrectly.)

### Steps to reproduce

to reproduce this bug:
1. play the CS:GO map, "dz_sirocco", in single-player sandbox mode
2. attempt a descent into this well on foot (don't noclip): 
![dz_sirocco0000](https://user-images.githubusercontent.com/10889213/81864864-6f7f4c00-952a-11ea-80f7-760b0a3b8dcb.jpg)
3. observe the strange collisions as you run into them on the way down

#### optional steps:
1. download BSP source from https://github.com/ata4/bspsrc/releases/tag/v1.4.0
2. use it to decompile the cs:go map "dz_sirocco"
3. open the decompiled map in a version of hammer that is properly configured for garry's mod mapping (but is still not gmod's own hammer)
4. look at that well and notice all the weird giant pipe models which are now in the exact positions that you ran into the invisible walls in-game. notice that in this screenshot i have quick-hidden some brushes that would otherwise obscure the view of the pipes: 
![image](https://user-images.githubusercontent.com/10889213/81865699-735f9e00-952b-11ea-8d63-c5c71aff012a.png)

**Comment by robotboy655** [comment]
Yes I am aware of this, the issue is that scaling the props is just way too slow, it would increase loading times significantly on the maps that have scaled props.

**Comment by Fennecai** [comment]
how is this for a suggestion then: don't do it on map load, but instead scale the props during gameplay based on how close they are to a physics object or player? that way the props will only be physically scaled when they need to be, and not when they don't. just a suggestion.

**Comment by Fennecai** [comment]
i had a better idea- what if the collision on scaled models was just disabled completely?

**Comment by Kefta** [comment]
That would be even worse.

**Comment by Fennecai** [comment]
> That would be even worse.

how? right now certain maps cant even be played due to random giant invisible walls as a result of this bug. with the collisions disabled you could at least play the maps decently well.

**Comment by Kefta** [comment]
Scaled props are used in some CS:GO maps for blocking off certain areas, which is still preserved when scaled up. I honestly don't know why the fix isn't merged regardless of load time increase since it severely impacts playability anyway.

**Comment by Fennecai** [comment]
both of those are dang good points.

**Comment by Fennecai** [comment]
when i said both are good points: 

"Scaled props are used in some CS:GO maps for blocking off certain areas, which is still preserved when scaled up. " true, but...

" it severely impacts playability anyway."

so for the sake of Garry's mod sandbox being probably more important than legitimate CS:GO gameplay when it comes to playing...well, gmod sandbox, why isn't this gonna be fixed?

or are you still thinking of it? this issue wasn't marked "cant/wontfix" or anything so I'm wondering what is going on in regards to this.

honestly if its about loading times, i'd argue that that shouldn't matter as quite a few people play with hundreds of addons installed, which also impacts load times just as much as fixing this might. the people who play gmod in near vanilla are few in number. of course the devs have the actual stats to look at, and i don't, so i might be wrong, but its worth a thought.

what if collisions were only removed for downscaled props? that way the intentional area blocking wouldn't be harmed by this unless the blockage was made stupidly by downscaling lots of props and then proceeding to build a wall out of them like bricks. i don't think a map that did that would become an official csgo map- very un-optimized way to do things. 

sorry if I'm being forceful or impatient or anything, i do realize that there are reasons why this isn't fixed yet probably, i am merely wondering what those reasons might be since there hasn't been any word on it.

**Comment by robotboy655** [comment]
Should be fixed on `dev` beta.

**Comment by robotboy655** [comment]
It has to be reverted because it without the faster code to scale the physics (which causes issues on its own, like infinite loops on `gm_sentimental98v1` Linux only with any players/bots on the server) it just crashes on maps like `dz_sirocco`. Needs to be revisited later somehow.
